src/coq_rules.ml

```diff
@@ -90,11 +120,42 @@ let create_ccoq sctx ~dir =
   ; coqpp  = rr "coqpp"
   }
 
+(* compute include flags and mlpack rules *)
+let setup_ml_deps ~scope ~stdlib_dir libs =
+
+  (* coqdep expects an mlpack file next to the sources otherwise it
+   * will omit the cmxs deps *)
+  let ml_pack_files lib =
+    let plugins = Mode.Dict.get (Lib.plugins lib) Mode.Native in
+    (* Should clean up once we can depend on >= 4.04 *)
+    let to_mlpack file =
+      Path.(of_string ((fst @@ Filename.split_extension (to_string file)) ^ ".mlpack")) in
```

Comment:
Could you add a set_extension functy to Path?

Reply:
I tried to backport from Ocaml as I mentioned below but the stuff gets a bit tricky. I could cook Dune's own of course

Reply:
Gets tricky due to platform-specific code.

Reply:
Note that your current implementation is fine with me. It's just that it's not nice to see this kind of ugly code in the middle of our actual logic.

Reply:
Indeed, I tried a quick addition to  `Path`, not sure how robust it is.

Reply:
I tweaked it a bit.